src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -164,6 +170,31 @@ entails SolverOptions{..} constraint context hints =
     forClassName ctx cn@(Qualified (Just mn) _) tys = concatMap (findDicts ctx cn) (ordNub (Nothing : Just mn : map Just (mapMaybe ctorModules tys)))
     forClassName _ _ _ = internalError "forClassName: expected qualified class name"
 
+    -- Left biased union of two row types
+    unionRows :: Type -> Type -> Type -> Maybe (Type, Type, Type, Maybe [Constraint])
+    unionRows l r o = if solverDeferErrors || not (hasKnown l || hasKnown r) then Nothing
+      else
+        let (ls, lEnd) = rowToList l
+            (rs, rEnd) = rowToList r
+            (_, oEnd) = rowToList o
+            lrUnion = ordNub (ls <> rs)
```

Comment:
No, those shouldn't unify. A row really is a `Map Symbol [Type]` at the type level now, so those two are different.